zebra/zebra_vty.c

```diff
@@ -2041,6 +2044,213 @@ DEFPY (show_route_summary,
+
+		if (nexthop->ifindex) {
+			vty_out(vty, "      interfaceIndex : %d\n",
+				nexthop->ifindex);
+			vty_out(vty, "      interfaceName : %s\n",
+				ifindex2ifname(nexthop->ifindex,
+					       nexthop->vrf_id));
+		}
+
+		if (nexthop->src.ipv4.s_addr) {
+			if (inet_ntop(AF_INET, &nexthop->src.ipv4, buf,
+				      sizeof(buf)))
+				vty_out(vty, "      source : %s\n", buf);
+		}
+		break;
+	case NEXTHOP_TYPE_IPV6:
+	case NEXTHOP_TYPE_IPV6_IFINDEX:
+		vty_out(vty, "      ip : %s\n",
+			inet_ntop(AF_INET6, &nexthop->gate.ipv6, buf,
+				  sizeof(buf)));
+		vty_out(vty, "      afi : ipv6\n");
+
+		if (nexthop->ifindex) {
+			vty_out(vty, "      interfaceIndex : %d\n",
+				nexthop->ifindex);
+			vty_out(vty, "      interfaceName : %s\n",
+				ifindex2ifname(nexthop->ifindex,
+					       nexthop->vrf_id));
+		}
+
+		if (!IPV6_ADDR_SAME(&nexthop->src.ipv6, &in6addr_any)) {
+			if (inet_ntop(AF_INET6, &nexthop->src.ipv6, buf,
+				      sizeof(buf)))
+				vty_out(vty, "      source : %s\n", buf);
+		}
+		break;
+	case NEXTHOP_TYPE_IFINDEX:
+		vty_out(vty, "      Nexthop is ifindex(directlyConnected).\n");
+		vty_out(vty, "      interfaceIndex : %d\n", nexthop->ifindex);
+		vty_out(vty, "      interfaceName : %s\n",
+			ifindex2ifname(nexthop->ifindex, nexthop->vrf_id));
+		break;
+	case NEXTHOP_TYPE_BLACKHOLE:
+		vty_out(vty,
+			"      Nexthop is Blackhole interface(unreachable).\n");
+
+		switch (nexthop->bh_type) {
+		case BLACKHOLE_REJECT:
+			vty_out(vty, "      Blackhole type : reject\n");
+			break;
+		case BLACKHOLE_ADMINPROHIB:
+			vty_out(vty,
+				"      Blackhole type : admin-prohibited\n");
+			break;
+		case BLACKHOLE_NULL:
+			vty_out(vty, "      Blackhole type : NULL0\n");
+			break;
+		case BLACKHOLE_UNSPEC:
+			break;
+		}
+		break;
+	default:
+		break;
+	}
+}
+
+static void show_ip_route_dump_vty(struct vty *vty, struct route_table *table)
+{
+	struct route_node *rn;
+	struct route_entry *re;
+	char buf[SRCDEST2STR_BUFFER];
+	char time[20];
+	time_t uptime;
+	struct tm *tm;
+	struct timeval tv;
+	struct nexthop *nexthop = NULL;
+	int nexthop_num = 0;
+
+	vty_out(vty, "\nIPv4/IPv6 Routing table dump\n");
+	vty_out(vty, "----------------------------\n");
+
+	for (rn = route_top(table); rn; rn = route_next(rn)) {
+		RNODE_FOREACH_RE (rn, re) {
+			vty_out(vty, "Route: %s\n",
+				srcdest_rnode2str(rn, buf, sizeof(buf)));
+			vty_out(vty, "   protocol : %s\n",
+				zebra_route_string(re->type));
+			vty_out(vty, "   instance : %u\n", re->instance);
+			vty_out(vty, "   vrfId : %u\n", re->vrf_id);
+			vty_out(vty, "   flags : %u\n", re->flags);
+
+			if (re->type != ZEBRA_ROUTE_CONNECT) {
+				vty_out(vty, "   distance : %u\n",
+					re->distance);
+				vty_out(vty, "   metric : %u\n", re->metric);
+			}
+
+			vty_out(vty, "   tag : %u\n", re->tag);
+
+			uptime = monotime(&tv);
+			uptime -= re->uptime;
+			tm = gmtime(&uptime);
+
+			if (uptime < ONE_DAY_SECOND)
+				snprintf(time, 20, "%02d:%02d:%02d",
```

Comment:
Do not hard code buffer sizes. Use `sizeof(time)`. This comment applies to subsequent uses as well.

Reply:
done